### PR TITLE
Documents improvements

### PR DESCRIPTION
**Pull request by TheDaChicken**
Deals with #346 

It wasn't straightforward what the sample rate was meant to be for either of the 2 pipe functions. This adds macro and documents it for each function. 

~~This also deals with a warning that Clion IDE gave me. It warned me that `__declspec(dllexport)` was meant to be used in the header. If needed, this part can be removed. This is a nick-pick. CMake newer versions can do this automatically using `set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)` though is not supported with this 2.8.~~

If you think of other constants that should be added, let me know. 

My program has a simple `#define NRSC5_MPS_PROGRAM 0x0` constant. Not sure if that is worth adding. It is simple zero.

### CODE REVIEW

File: support/nrsc5.py
Comment (argilo) [inline support/nrsc5.py:35]: Grouping these into an `Enum` seems clumsy, since it takes extra work to get at the underlying values. Maybe it would be easier to make them contants:

```suggestion
SAMPLE_RATE_CU8 = 1488375
SAMPLE_RATE_CS16 = 744187.5
SAMPLE_RATE_AUDIO = 44100
```

**Comment by argilo** [comment]
> This also deals with a warning that Clion IDE gave me. It warned me that `__declspec(dllexport)` was meant to be used in the header. If needed, this part can be removed.

If you don't mind, I'd rather consider that in a separate PR.

**Comment by TheDaChicken** [comment]
> > This also deals with a warning that Clion IDE gave me. It warned me that `__declspec(dllexport)` was meant to be used in the header. If needed, this part can be removed.
> 
> If you don't mind, I'd rather consider that in a separate PR.

Alright, I created, #350. Other than the review, it should be all done!

File: src/defines.h
Comment (argilo) [inline src/defines.h:12]: Do you suppose we should get rid of `SAMPLE_RATE` and just use `NRSC5_SAMPLE_RATE_CU8` instead?

**Review by argilo — APPROVED** [review]
Looks good now. I'll merge once CI is happy.

**Comment by TheDaChicken** [comment]
Also a few things before merge:

Are you sure that it decimates only 2 from cs16? What about AM? There is DECIMATION_FACTOR_AM in acquire.c and also looks like it decimates by 32 in nrsc5.c 

Should there be an NRSC5_PACKET_SAMPLES constant to prevent the 2048 magic number in main.c?

**Comment by argilo** [comment]
> Are you sure that it decimates only 2 from cs16? What about AM? There is DECIMATION_FACTOR_AM in acquire.c and also looks like it decimates by 32 in nrsc5.c

That's a good point. For AM, there are five cascaded decimators, so the CS16 sample rate is 1488375 / 32 = 46511.71875. It would make sense to have separate constants for FM and AM.

> Should there be an NRSC5_PACKET_SAMPLES constant to prevent the 2048 magic number in main.c?

Yes, I think that would be nice. They are called audio frames in the specification, so perhaps the name could be `NRSC5_AUDIO_FRAME_SAMPLES`.

**Comment by TheDaChicken** [comment]
> > Are you sure that it decimates only 2 from cs16? What about AM? There is DECIMATION_FACTOR_AM in acquire.c and also looks like it decimates by 32 in nrsc5.c
> 
> That's a good point. For AM, there are five cascaded decimators, so the CS16 sample rate is 1488375 / 32 = 46511.71875. It would make sense to have separate constants for FM and AM.
> 
> > Should there be an NRSC5_PACKET_SAMPLES constant to prevent the 2048 magic number in main.c?
> 
> Yes, I think that would be nice. They are called audio frames in the specification, so perhaps the name could be `NRSC5_AUDIO_FRAME_SAMPLES`.

Alright. Those have been added!

**Pull request merged**
